src/scripts/term_range_expansion.py

```diff
@@ -0,0 +1,207 @@
+"""Takes in the full ontology and the exclusions tsv and extracts a simple list of terms from it.
+Outputs two files:
+1. a simple list of terms (config/ONTO_NAME_term_exclusions.txt)
+2. a simple two column file with the term_id and the exclusion reason
+
+# Resources
+- GitHub issue: https://github.com/monarch-initiative/mondo-ingest/issues/22
+
+TODO's
+  - repurpose this: (i) to be generalized, (ii) to do the job of consuming and producing exclusion table as described in issue
+  - possible bug: ICD10CM:C7A-C7A reported as term, but should be ICD10CM:C7A
+"""
+import os
+import subprocess
+from argparse import ArgumentParser
+from copy import copy
+from typing import Any, Dict, List
+
+from jinja2 import Template
+import pandas as pd
+
+
+# Vars
+# # Config
+USE_CACHE = False
+
+# # Static
+PROJECT_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..')
+SPARQL_DIR = os.path.join(PROJECT_DIR, 'sparql')
+CACHE_DIR = os.path.join(PROJECT_DIR, 'cache')
+sparql_child_inclusion_path = os.path.join(PROJECT_DIR, 'sparql', 'get-terms-children.sparql.jinja2')
+sparql_child_exclusion_path = os.path.join(PROJECT_DIR, 'sparql', 'get-terms.sparql.jinja2')
+input_exclusion_term_ranges_csv_path = os.path.join(
+    PROJECT_DIR, 'preparation', 'icd10cm_excluded_term_ranges_intensional.csv')
+input_icd10cm_ttl_path = os.path.join(PROJECT_DIR, 'preparation', 'icd10cm.ttl')
+TERM_RANGE_MAX_VAL = 99
+
+
+# Functions
+# todo: pd.errors.EmptyDataError: I think this happens because Python process needs to stop before results saved?
+def sparql_jinja2_file_query__via_robot(
```

Comment:
I have copy/pasted and repurposed this function many times in several repositories. I'm guessing that it would be good to standardize this and move it into some library. Maybe OAK?